src/status_im/navigation/effects.cljs

```diff
@@ -17,37 +16,36 @@
 
 (defn get-status-nav-color
   [view-id theme]
-  (let [theme (or (get-in views/screens [view-id :options :theme])
-                  theme)
-        [rnn-status-bar rn-status-bar]
-        (if (or (= theme :dark)
-                @state/alert-banner-shown?
-                (and (= view-id :shell-stack) (not (jump-to.utils/home-stack-open?))))
-          [:light "light-content"]
-          [:dark "dark-content"])
-        home-stack? (some #(= view-id %) shell.constants/stacks-ids)
+  (let [theme            (or (get-in views/screens [view-id :options :theme])
+                             theme)
+        status-bar-theme (if (or (= theme :dark)
+                                 @state/alert-banner-shown?
+                                 (and (= view-id :shell-stack)
+                                      (not (jump-to.utils/home-stack-open?))))
+                           :light
+                           :dark)
+        home-stack?      (some #(= view-id %) shell.constants/stacks-ids)
         ;; Home screen nav bar always dark due to bottom tabs
-        nav-bar-color (if (or home-stack?
-                              (= view-id :shell-stack)
-                              (= theme :dark))
-                        colors/neutral-100
-                        colors/white)
-        comp-id (if (or home-stack?
-                        (jump-to.utils/shell-navigation? view-id)
-                        (= view-id :shell))
-                  :shell-stack
-                  view-id)]
-    [rnn-status-bar rn-status-bar nav-bar-color comp-id]))
+        nav-bar-color    (if (or home-stack?
+                                 (= view-id :shell-stack)
+                                 (= theme :dark))
+                           colors/neutral-100
+                           colors/white)
+        comp-id          (if (or home-stack?
+                                 (jump-to.utils/shell-navigation? view-id)
+                                 (= view-id :shell))
+                           :shell-stack
+                           view-id)]
+    [status-bar-theme nav-bar-color comp-id]))
 
 (defn reload-status-nav-color-fx
   [[view-id theme]]
   (when (and (= @state/root-id :shell-stack) view-id)
-    (let [[rnn-status-bar rn-status-bar nav-bar-color comp-id] (get-status-nav-color view-id theme)]
-      (if platform/ios?
-        (rn/set-status-bar-style rn-status-bar true)
```

Comment:
removed update of status-bar color using rn. It only worked for simulator https://github.com/status-im/status-mobile/issues/20495#issuecomment-2186858868